api/dao/hierarchy.py

```diff
@@ -131,6 +131,24 @@ def get_parent_tree(cont_name, _id):
 
     return tree
 
+def propagate_group_change(_id, update, oper):
+    # Apply change to projects
+    if oper == 'POST':
+        config.db.projects.update_many({'group': _id}, {'$addToSet': {'permissions': update}})
+    elif oper == 'PUT':
+        config.db.projects.update_many({'group': _id, 'permissions._id' : update['_id']}, {'$set': {'permissions.$.access': update['access']}})
+    elif oper == 'DELETE':
+        config.db.projects.update_many({'group': _id, 'permissions._id' : update}, {'$pull' : {'permissions': {'_id': update}}})
+    else:
+        raise APIStorageException("Cannot propagate {} operation".format(oper))
```

Comment:
Could you reuse the function below and set the `query` and `update` commands based on what you would do here? I'd prefer reusing complicated hierarchy functions when possible.